### PR TITLE
Update auth tests definition

### PR DESCRIPTION
**Pull request by yang-g**
- removed service_account_creds test case
- made username check stricter to avoid empty username working as passed
- changed oauth2_auth_token so that both service account and GCE creds can be used
- changed per_rpc_creds to use JWT credentials instead of ServiceAccountCreds

### CODE REVIEW

File: doc/interop-test-descriptions.md [hunk 1]
```diff
@@ -582,30 +538,32 @@ Procedure:
...
+by the auth library. The client can optionally check the username matches the
+email address in the key file.
 * response payload body is 314159 bytes in size
 * clients are free to assert that the response payload body contents are zero
   and comparing the entire response message against a golden response
 
 ### oauth2_auth_token
 
-Similar to the other auth tests, this test is only for cloud-to-prod path.
+This test is only for cloud-to-prod path and runs in GCE only.
```
Comment (ejona86) [inline doc/interop-test-descriptions.md]: and _some implementations_ may run in GCE only

**Comment by ejona86** [comment]
@yang-g, it LGTM. One minor tweak and then I'll merge.

**Comment by yang-g** [comment]
@ejona86 Done. PTAL Thanks.

**Comment by yang-g** [comment]
@tbetbetbe @iamqizhao @murgatroid99 @stanley-cheung @nathanielmanistaatgoogle heads up
@jtattermusch reminded me this.

**Comment by jtattermusch** [comment]
Two things:
It turns out that with the current API, C# doesn't support combination of JWT access token and per-rpc creds. Because without using the per-stub interceptor, I have no way of finding out what the authUri (JWT token audience is).  I think this is going to be the same for other wrapped languages as well.

Is there a reason why jwt_token_creds test doesn't allow using application default credentials and requires me to load service account from a file directly? I think using app default credentials is a more natural choice for me here.

**Comment by ejona86** [comment]
@jtattermusch, why can't you use the per-stub interceptor? Which test is that in reference to?

**Comment by yang-g** [comment]
I think the last jwt_token_creds should be per_rpc_creds, which is the case
Jan was talking about.

**Comment by jtattermusch** [comment]
#3262

**Comment by yang-g** [comment]
@ejona86 Maybe relax per_call_creds to allow auth_library native creds on the client?

**Comment by yang-g** [comment]
@ejona86 never mind. @jtattermusch mentioned in #3262 that eventually the current spec will work out for him.

**Pull request merged**